### PR TITLE
COM-465 - Horizontal form layout

### PR DESCRIPTION
**Pull request by andrearutrecht**
The branch is based on `feature/refactor-admin-component-theming`.

Horizontal layout for form fields now only takes effect, when the field exceeds 600px width.

<details><summary>Video:</summary>
<p>

https://github.com/vivid-planet/comet/assets/122883866/77c5e755-0d83-419a-899f-fd09b4762c68

</p>
</details>

### CODE REVIEW

File: .changeset/khaki-eels-try.md
Comment (johnnyomair) [inline .changeset/khaki-eels-try.md:5]: This is an internal change which isn't needed in the changelog

File: .changeset/khaki-eels-try.md
Comment (johnnyomair) [inline .changeset/khaki-eels-try.md:7]:
```suggestion
Only use horizontal layout in `FieldContainer` when it exceeds `600px` in width
```

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/form/FieldContainer.tsx:93]: What is this arbrtrary number?

**Review by johnnyomair — COMMENTED** [review]
Can't we use a CSS container query for this?

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (andrearutrecht) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: This number is mentioned in the ticket like that.

Comment (johnnyomair) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: @jamesricky should this be part of the theme?

**Comment by johnnyomair** [comment]
@andrearutrecht the task mentions a gap of 20px which is missing here. Also, this should be done via a CSS container query.

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (jamesricky) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: This number was defined by UX, I think it's the width form's have, when they are inside a Dialog, with the default size. They decided to make this the maximum width in general. 

I don't think we need to do anything in the theme for that, for the rare case, that this should be `100%` or something else, we can use the `styleOverrides` in the project's theme to change this.

**Comment by andrearutrecht** [comment]
@Obi I'll add the gap. I've talked with @jamesricky about using a container query. Container queries cannot be used within the container they are defined, here it is `Root`. Ricky and I decided, that instead of adding another `div` layer, we calculate the width and use `forceVertical` for styling instead.

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (johnnyomair) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: If this is the max width in general, then shouldn't it also be applied for `variant="vertical"`? And should we add a comment or define a constant?

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (johnnyomair) [inline packages/admin/admin/src/form/FieldContainer.tsx:89]: I believe this is only related to container queries, so it can be removed, right?

**Comment by johnnyomair** [comment]
> Ricky and I decided, that instead of adding another `div` layer, we calculate the width and use `forceVertical` for styling instead.

IMHO an additional div and a CSS-only solution would be preferable to a JS solution, but I'll trust @jamesricky's judgement here

File: packages/admin/admin/src/form/FieldContainer.tsx
Comment (jamesricky) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: > If this is the max width in general, then shouldn't it also be applied for variant="vertical"?

This would cause an issue where the fields would unintentionally be rendered next to each other, if the form/window is twice the width of the fields, as they have `display: inline-flex` unlike the `display: flex` of the `horizontal` variant. 

But according to the UX-Team, the `vertical` variant might be phased out in general, as with this PR the fields are shown in a more space-efficient way, depending on the width they have available. 
So it might not be worth the effort to optimize the `vertical` variant.

Comment (jamesricky) [reply packages/admin/admin/src/form/FieldContainer.tsx:93]: > And should we add a comment or define a constant?

I don't think this is necessary, as it's defined in the UX-Teams design guidelines 🤷🏼

**Comment by jamesricky** [comment]
> IMHO an additional div and a CSS-only solution would be preferable to a JS solution, but I'll trust @jamesricky's judgement here

@johnnyomair in general I'd prefer the additional div as well but I fear this would break applications that have custom styles for the `FieldContainer`... not sure if it's a good idea to add this breaking change 🤔

**Comment by johnnyomair** [comment]
@jamesricky do you want to merge this in the theme refactor feature branch, or should we wait until it has been merged?

**Comment by jamesricky** [comment]
> @jamesricky do you want to merge this in the theme refactor feature branch, or should we wait until it has been merged?

IMO we should wait and merge it into `next`

**Review by jamesricky — APPROVED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**
